TESTS/events/queue/main.cpp

```diff
@@ -0,0 +1,267 @@
+#include "mbed_events.h"
+#include "mbed.h"
+#include "rtos.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+
+using namespace utest::v1;
+
+
+// flag for called
+volatile bool touched = false;
+
+// static functions
+void func5(int a0, int a1, int a2, int a3, int a4) { 
+    touched = true;
+    TEST_ASSERT_EQUAL(a0 | a1 | a2 | a3 | a4, 0x1f);
+}
+
+void func4(int a0, int a1, int a2, int a3) {
+    touched = true;
+    TEST_ASSERT_EQUAL(a0 | a1 | a2 | a3, 0xf); 
+}
+
+void func3(int a0, int a1, int a2) {
+    touched = true;
+    TEST_ASSERT_EQUAL(a0 | a1 | a2, 0x7);
+}
+
+void func2(int a0, int a1) {
+    touched = true;
+    TEST_ASSERT_EQUAL(a0 | a1, 0x3);
+}
+
+void func1(int a0) {
+    touched = true;
+    TEST_ASSERT_EQUAL(a0, 0x1);
+}
+
+void func0() {
+    touched = true;
+}
+
+#define SIMPLE_POSTS_TEST(i, ...)                           \
+void simple_posts_test##i() {                               \
+    EventQueue queue;                                       \
+                                                            \
+    touched = false;                                        \
+    queue.call(func##i,##__VA_ARGS__);                      \
+    queue.dispatch(0);                                      \
+    TEST_ASSERT(touched);                                   \
+                                                            \
+    touched = false;                                        \
+    queue.call_in(1, func##i,##__VA_ARGS__);                \
+    queue.dispatch(2);                                      \
+    TEST_ASSERT(touched);                                   \
+                                                            \
+    touched = false;                                        \
+    queue.call_every(1, func##i,##__VA_ARGS__);             \
+    queue.dispatch(2);                                      \
+    TEST_ASSERT(touched);                                   \
+}
+
+SIMPLE_POSTS_TEST(5, 0x01, 0x02, 0x04, 0x08, 0x010)
+SIMPLE_POSTS_TEST(4, 0x01, 0x02, 0x04, 0x08)
+SIMPLE_POSTS_TEST(3, 0x01, 0x02, 0x04)
+SIMPLE_POSTS_TEST(2, 0x01, 0x02)
+SIMPLE_POSTS_TEST(1, 0x01)
+SIMPLE_POSTS_TEST(0)
+
+
+void time_func(Timer *t, int ms) {
+    TEST_ASSERT_INT_WITHIN(2, ms, t->read_ms());
+    t->reset();
+}
+
+template <int N>
+void call_in_test() {
+    Timer tickers[N];
+
+    EventQueue queue;
+
+    for (int i = 0; i < N; i++) {
+        tickers[i].start();
+        queue.call_in((i+1)*100, time_func, &tickers[i], (i+1)*100);
+    }
+
+    queue.dispatch(N*100);
+}
+
+template <int N>
+void call_every_test() {
+    Timer tickers[N];
+
+    EventQueue queue;
+
+    for (int i = 0; i < N; i++) {
+        tickers[i].start();
+        queue.call_every((i+1)*100, time_func, &tickers[i], (i+1)*100);
+    }
+
+    queue.dispatch(N*100);
+}
+
+struct big { char data[1024]; } big;
```

Comment:
It looks like the test issue is with this allocation. It looks like IAR's linker can't find a contiguous region of space?

We may want to omit this test (allocate_failure_test1) for now. It has had other problems such as overflowing stacks, and allocation failure is also tested by allocat_failure_test2.